Source/Pawnmorphs/Esoteria/FormerHumanUtilities.cs

```diff
@@ -123,15 +123,74 @@ static FormerHumanUtilities()
+        /// <param name="allowRestricted">Include restricted animals.</param>
+		/// <returns>
+		///   <c>true</c> if [is valid former human] [the specified forced]; otherwise, <c>false</c>.
+		/// </returns>
+		public static bool IsValidFormerHuman(this ThingDef pawnKind, bool allowRestricted = true, bool allowDisabled = false)
+        {
+            if (pawnKind.race == null)
+                return false;
 
-            _allRegularFormerHumanPawnKinds =
-                DefDatabase<PawnKindDef>.AllDefsListForReading.Where(p => p.RaceProps?.Animal == true
-                                                                       && p.race.GetModExtension<ChaomorphExtension>() == null
-                                                                       && p.race.GetModExtension<FormerHumanSettings>()
-                                                                          ?.neverFormerHuman
-                                                                       != true)
-                                        .ToList(); 
-        }
+            // Must be an animal
+            if (pawnKind.race.Animal == false)
+                return false;
+
+            // Dryads are not controllable 
+            if (pawnKind.race.Dryad)
+                return false;
+
+            // XML Patched as not valid former human.
+            if (pawnKind.GetModExtension<FormerHumanSettings>()?.neverFormerHuman == true)
+                return false;
+
+            // If restricted is not allowed, then filter away special chaomorphs.
+            ChaomorphExtension chaoExtension = pawnKind.GetModExtension<ChaomorphExtension>();
+			if (chaoExtension != null && allowRestricted == false)
+            {
+                // Only ordinary chaomorphs are allowed normally.
+                if (chaoExtension.chaoType != ChaomorphType.Chaomorph)
+                    return false;
+
+                // Chamorphs with a selection weight of less than 0 is not allowed.
+                if (chaoExtension.selectionWeight < 0)
+                    return false;
+            }
+
+            // Check if manually blocked
+            if (PawnmorpherMod.Settings.animalBlacklist.TryGetValue(pawnKind.defName, out FormerHumanRestrictions restriction))
+            {
+                if (allowRestricted == false && restriction == FormerHumanRestrictions.Restricted)
+                    return false;
+
+				if (allowDisabled == false && restriction == FormerHumanRestrictions.Disabled)
+					return false;
+			}
+
+            return true;
+		}
+
+		/// <summary>
+		/// Caches the valid former humans. See <see cref="AllRegularFormerHumanPawnkindDefs"/>
+		/// </summary>
+		public static void CacheValidFormerHumans()
+        {
+            _allRegularFormerHumanPawnKinds.Clear();
+            _allRegularFormerHumanPawnKinds.AddRange(DefDatabase<PawnKindDef>.AllDefsListForReading.Where(p => p.race.IsValidFormerHuman(allowRestricted: false)));
```

Comment:
I have actually been considering this, if it would make more sense to filter on Pawnkinddef instead.

Though the difference is minimal in most places as it tend to be a 1:1 for animals, Martens from what I remember has multiple variations of pawnkinds of the same Marten species (ThingDef).
And in that case, you can't restrict only some of the kinds.

I imagine that it would be similar to having a wolf species that contains both "Pack member" and "Alpha", maybe "Apex".

Then you can't restrict it to only allow "Pack member" as former human.